eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -225,7 +225,7 @@ class EclairImpl(appKit: Kit) extends Eclair with Logging {
   } yield peerinfos
 
   override def node(nodeId: PublicKey)(implicit timeout: Timeout): Future[Option[Router.PublicNode]] = {
-    (appKit.router ? Router.GetNode(nodeId)).mapTo[Router.GetNodeResponse].map {
+    appKit.router.toTyped.ask(ref => Router.GetNode(ref, nodeId))(timeout, appKit.system.scheduler.toTyped).mapTo[Router.GetNodeResponse].map {
```

Comment:
```suggestion
    appKit.router.toTyped.ask(ref => Router.GetNode(ref, nodeId)).map {
```

Reply:
Hmm, I still still get an error: `No implicits found for scheduler: Scheduler' when I add:

```scala
import akka.actor.typed.scaladsl.AskPattern.{Askable, schedulerFromActorSystem}
```

Unless I do this:
```scala
  override def node(nodeId: PublicKey)(implicit timeout: Timeout): Future[Option[Router.PublicNode]] = {
    appKit.router.toTyped.ask(ref => Router.GetNode(ref, nodeId))(timeout, appKit.system.scheduler.toTyped).map {
      case n: PublicNode => Some(n)
      case _: UnknownNode => None
    }
  }
```
I think because the scheduler in the current context is untyped, the typed`schedulerFromActorSystem` isn't used.

Any other tips for resolving this? it seems the rest of the code also includes that boilerplate implicit scheduler.

Reply:
I guess you also need `akka.actor.typed.scaladsl.adapter.ClassicActorSystemOps`

Reply:
If I add this at the top of the class it works, and I can remove the boiler plate from a few other places.
```scala
class EclairImpl(appKit: Kit) extends Eclair with Logging {

  implicit val ec: ExecutionContext = appKit.system.dispatcher
  implicit val scheduler: Scheduler = appKit.system.scheduler.toTyped
...
  override def node(nodeId: PublicKey)(implicit timeout: Timeout): Future[Option[Router.PublicNode]] = {
    val asker = appKit.router.toTyped.ask(ref => Router.GetNode(ref, nodeId))
    asker .mapTo[Router.GetNodeResponse].map {
      case n: PublicNode => Some(n)
      case _: UnknownNode => None
    }
  }
...

Reply:
[c110900](https://github.com/ACINQ/eclair/pull/2575/commits/c1109005de2bb00926e27ed03bdf071fa50d98ec) fixes this and cleans up a few other instances where a typed implicit scheduler was required

Reply:
> akka.actor.typed.scaladsl.adapter.ClassicActorSystemOps

I'll give that a try

Reply:
@thomash-acinq I still don't get the implicit with these imports:

```
import akka.actor.typed.scaladsl.AskPattern.{Askable, schedulerFromActorSystem}
import akka.actor.typed.scaladsl.adapter.{ClassicActorRefOps, ClassicSchedulerOps}
import akka.actor.typed.scaladsl.adapter.ClassicActorSystemOps
```

Reply:
What you did looks good. It's much cleaner now.